OpenChange/MAPIStoreSamDBUtils.m

```diff
@@ -29,8 +29,16 @@
 
 #import "MAPIStoreSamDBUtils.h"
 
+
+/* safe_ldb_search signature */
+int safe_ldb_search(struct ldb_context *ldb, TALLOC_CTX *mem_ctx,
+		    struct ldb_result **result, struct ldb_dn *base,
+		    enum ldb_scope scope, const char * const *attrs,
+		    const char *exp_fmt, ...);
```

Comment:
I'd rather prefer to publish the function in `libmapiproxy.h` as it is safer and more clear. Although this requires changes in OpenChange PR.